test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterTests.cs

```diff
@@ -41,9 +40,9 @@ public void PrometheusExporterTest1()
                 metricsHttpServer.Start();
                 var label1 = new List<KeyValuePair<string, string>>();
                 label1.Add(new KeyValuePair<string, string>("dim1", "value1"));
-                metrics.Add(new Metric("ns", "metric1", "desc", label1, 100));
-                metrics.Add(new Metric("ns", "metric1", "desc", label1, 100));
-                metrics.Add(new Metric("ns", "metric1", "desc", label1, 100));
+                metrics.Add(new Metric<long>("ns", "metric1", "desc", label1, AggregationType.LONGSUM));
```

Comment:
If we didn't expose Metric using generics, and had concrete LongMetric & DoubleMetric, we wouldn't need to provide the AggregationType like we need to here.

What would happen if you created a `Metric<long>` and set AggregationType to `DoubleSum`?

Reply:
Even if we have separate LongMetric and DoubleMetric, I think we'd still need to do aggregation type LongSumData and DoubleSumData.

Users are not expected to create `Metric` directly - they only go through one of the instruments. (`Metric` class is still public - I'll probably need to adjust access modifier of `Metric` ctor to internal.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/issues/409#issuecomment-597758058